### PR TITLE
ARROW-6520: [Python] More consistent handling of specified schema when creating Table

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
https://issues.apache.org/jira/browse/ARROW-6520

Mainly added extra tests trying to cover the different cases.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
OK, updated this (in the end went with `names` first), and added some more tests (also for passing schema in `from_pandas`).

**Comment by jorisvandenbossche** [comment]
More comments on this? 
(for me it is ready to be merged I think)

File: python/pyarrow/array.pxi [hunk 1]
```diff
@@ -225,7 +225,7 @@ def asarray(values, type=None):
     -------
     arr : Array
     """
-    if isinstance(values, Array):
+    if isinstance(values, (Array, ChunkedArray)):
         if type is not None and not values.type.equals(type):
             values = values.cast(type)
```
[the remark this answers is not in the document]
Comment (wesm) [reply python/pyarrow/array.pxi]: I think it's okay

**Comment by wesm** [comment]
Looking

**Review by wesm — APPROVED** [review]
+1, thanks for sifting through these issues carefully and adding these tests =)

**Pull request closed without merging**